src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFileSystem.cs

```diff
@@ -243,8 +243,12 @@ public void AddFile(string path, MockFileData mockFile)
             }
 
             var directoryPath = Path.GetDirectoryName(fixedPath);
-
-            if (!DirectoryExistsWithoutFixingPath(directoryPath))
+            if (directoryPath == null)
+            {
+                AddDrive(fixedPath, new MockDriveData());
+                SetEntry(fixedPath, mockFile);
```

Comment:
Now you set the entry twice.
This line should be removed.

Reply:
oh my bad, fixed